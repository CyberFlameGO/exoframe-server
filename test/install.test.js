/* eslint-env jest */
const util = require('util');
const exec = util.promisify(require('child_process').exec);

// set timeout to 0.5s
jest.setTimeout(500);

const shellOptions = {
  env: {HOME: '/root'},
};

describe('Test install script', () => {
  test('Should print help', async () => {
    const {stdout} = await exec('tools/install.sh -h');
    expect(stdout).toMatchSnapshot();
  });

  test('Should print only docker command', async () => {
    const {stdout} = await exec('tools/install.sh --dry-run --password PASSWORD -d EXAMPLE.COM', shellOptions);
    expect(stdout).toMatchSnapshot();
  });

  test('Should include mkdir command', async () => {
    const {stdout} = await exec(
      'tools/install.sh --dry-run --password PASSWORD -d EXAMPLE.COM -e EMAIL@GMAIL.COM',
      shellOptions
    );
    expect(stdout).toMatchSnapshot();
  });

  test('Should fail command', async () => {
    expect(exec('tools/install.sh -fail')).rejects.toThrow('Command failed: tools/install.sh -fail', shellOptions);
  });
});
