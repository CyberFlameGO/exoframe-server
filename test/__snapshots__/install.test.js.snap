// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test install script Should include mkdir command 1`] = `
"
Commands to run inside server:

mkdir -p /root/.exoframe && touch /root/.exoframe/server.config.yml
echo \\"letsencrypt: true\\" >> /root/.exoframe/server.config.yml
echo \\"letsencryptEmail: EMAIL@GMAIL.COM\\" >> /root/.exoframe/server.config.yml


docker run -d -v /var/run/docker.sock:/var/run/docker.sock -v /root/.exoframe:/root/.exoframe -v /root/.ssh/authorized_keys:/root/.ssh/authorized_keys:ro -e EXO_PRIVATE_KEY=PASSWORD --label traefik.enable=true --label traefik.http.routers.exoframe-server.rule=Host(\`exoframe.EXAMPLE.COM\`) --label traefik.http.routers.exoframe-server-web.rule=Host(\`exoframe.EXAMPLE.COM\`) --label traefik.http.routers.exoframe-server.tls.certresolver=exoframeChallenge --label traefik.http.middlewares.exoframe-server-redirect.redirectscheme.scheme=https --label traefik.http.routers.exoframe-server-web.entrypoints=web --label traefik.http.routers.exoframe-server-web.middlewares=exoframe-server-redirect@docker --label traefik.http.routers.exoframe-server.entrypoints=websecure --label entryPoints.web.address=:80 --label entryPoints.websecure.address=:443 --restart always --name exoframe-server exoframe/server
"
`;

exports[`Test install script Should print help 1`] = `
"
Usage:
  -D, --dry-run     Dry run. Print command instead of executing it.
  -e, --email       Enter email to enable SSL support.
  -d, --domain      * Enter exoframe-server domain.
  -p, --password    * Enter your private key used for JWT encryption.

"
`;

exports[`Test install script Should print only docker command 1`] = `
"
Commands to run inside server:

docker run -d -v /var/run/docker.sock:/var/run/docker.sock -v /root/.exoframe:/root/.exoframe -v /root/.ssh/authorized_keys:/root/.ssh/authorized_keys:ro -e EXO_PRIVATE_KEY=PASSWORD --label traefik.enable=true --label traefik.http.routers.exoframe-server.rule=Host(\`exoframe.EXAMPLE.COM\`) --restart always --name exoframe-server exoframe/server
"
`;
